Author: Stas Boukarev <stassats@gmail.com>  2022-11-29 13:45:01
Committer: Stas Boukarev <stassats@gmail.com>  2022-11-29 14:06:01
Parent: cacbb24d03cbc1309447bfcd72a5cd17dd2d0e8b (Minor manual enhancements)
Branches: master, remotes/origin/master
Follows: sbcl-2.2.11
Precedes: 

    Fix delete-arena.
    
    Needs to use a nonvolatile register. (Why isn't ALIEN-FUNCALL used?)
    Also fix a style warning reported by Andreas Wacknitz.

--- sbcl-2.2.11/src/code/arena.lisp.orig	2022-11-28 20:47:28.000000000 +0000
+++ sbcl-2.2.11/src/code/arena.lisp	2022-11-29 19:37:50.474042566 +0000
@@ -91,7 +91,8 @@
 (define-vop (delete-arena)
   (:args (x :scs (descriptor-reg)))
   (:temporary (:sc unsigned-reg :offset rdi-offset :from (:argument 0)) rdi)
-  (:temporary (:sc unsigned-reg) rsp-save)
+  #+immobile-space
+  (:temporary (:sc unsigned-reg :offset rbx-offset) rsp-save)
   (:vop-var vop)
   (:generator 1
     (move rdi x)
--- sbcl-2.2.11/tests/arena.impure.lisp.orig	2022-11-28 20:47:28.000000000 +0000
+++ sbcl-2.2.11/tests/arena.impure.lisp	2022-11-29 19:28:30.246995301 +0000
@@ -218,8 +218,7 @@
               est act frac)
       (assert (< frac 1))))))
 
-(test-util:with-test (:name :thread-arena-inheritance
-                      :skipped-on :darwin) ; I can't imagine why this fails
+(test-util:with-test (:name :thread-arena-inheritance)
   (sb-vm:with-arena (*arena*)
     (let ((thread
            (sb-thread:make-thread
